pkg/cmd/repo/sync/sync.go

```diff
@@ -0,0 +1,286 @@
+package sync
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"regexp"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/git"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/internal/run"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/safeexec"
+	"github.com/spf13/cobra"
+)
+
+type SyncOptions struct {
+	HttpClient    func() (*http.Client, error)
+	IO            *iostreams.IOStreams
+	BaseRepo      func() (ghrepo.Interface, error)
+	Remotes       func() (context.Remotes, error)
+	CurrentBranch func() (string, error)
+	DestArg       string
+	SrcArg        string
+	Branch        string
+	Force         bool
+	SkipConfirm   bool
+}
+
+func NewCmdSync(f *cmdutil.Factory, runF func(*SyncOptions) error) *cobra.Command {
+	opts := SyncOptions{
+		HttpClient:    f.HttpClient,
+		IO:            f.IOStreams,
+		BaseRepo:      f.BaseRepo,
+		Remotes:       f.Remotes,
+		CurrentBranch: f.Branch,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "sync [<destination-repository>]",
+		Short: "Sync a repository",
+		Long: heredoc.Doc(`
+			Sync destination repository from source repository.
+
+			Without an argument, the local repository is selected as the destination repository.
+			By default the source repository is the parent of the destination repository.
+			The source repository can be overridden with the --source flag.
+		`),
+		Example: heredoc.Doc(`
+			# Sync local repository from remote parent
+			$ gh repo sync
+
+			# Sync local repository from remote parent on non-default branch
+			$ gh repo sync --branch v1
+
+			# Sync remote fork from remote parent
+			$ gh repo sync owner/cli-fork
+
+			# Sync remote repo from another remote repo
+			$ gh repo sync owner/repo --source owner2/repo2
+		`),
```

Comment:
I think this command has potential to be confusing, do people think this description and examples are clear and explanatory enough?